src/LiquidityPool.sol

```diff
@@ -210,12 +234,18 @@ contract LiquidityPool is Auth {
         return share.balanceOf(owner);
     }
 
-    function transfer(address recipient, uint256 amount) public auth returns (bool) {
-        return share.transferFrom(msg.sender, recipient, amount);
+    function transferFrom(address sender, address recipient, uint256 amount)
+        public
+        withTokenApproval(sender, amount)
+        returns (bool)
+    {
+        // discuss if we should add this here
+        // approveForOwner(sender, address(this), amount);
```

Comment:
Since we're checking for approvals now, I think we can remove this, right?

Reply:
nope, that would be an additional approval that we need to discuss